lib/node_modules/@stdlib/math/base/special/ellipj/lib/assign.js

```diff
@@ -0,0 +1,237 @@
+* // returns [ NaN, NaN, NaN, NaN ]
+*/
+function assign(u, m, out, stride, offset) {
+	var dnDenom;
+	var uK2cen;
+	var k1inv;
+	var sechu;
+	var sinhu;
+	var tanhu;
+	var phi0;
+	var phi1;
+	var atmp;
+	var FLG;
+	var gdu;
+	var uK2;
+	var uK4;
+	var mu;
+	var K2;
+	var K4;
+	var u0;
+	var sn;
+	var cn;
+	var dn;
+	var am;
+	var k;
+	var s;
+	var c;
+	var f;
+	var a;
+	var b;
+	var N;
+
+	if ( m < 0.0 ) {
+		// A&S 16.10.1 for a negative parameter, mapping -m to 0 < mu < 1
+		mu = -m / ( 1.0 - m );
+		k1inv = sqrt( 1.0 - m );
+		assign( u * k1inv, mu, tmp4, 1, 0 );
+		sn = ( tmp4[ 0 ] / tmp4[ 2 ] ) / k1inv;
+		cn = tmp4[ 1 ] / tmp4[ 2 ];
+		dn = 1.0 / tmp4[ 2 ];
+		am = NaN;
+	} else if ( m > 1.0 ) {
+		// A&S 16.11.1 for reciprocal parameter, mapping m > 1 to 0 < mu < 1
+		k = sqrt( m );
+		assign( u * k, 1.0 / m, tmp4, 1, 0 );
+		sn = tmp4[ 0 ] / k;
+		cn = tmp4[ 2 ];
+		dn = tmp4[ 1 ];
+		am = NaN;
+	} else if ( m === 0.0 ) {
+		// A&S table 16.6, limiting case m = 0: circular trigonometric functions
+		sincos( u, tmp2, 1, 0 );
+		sn = tmp2[ 0 ];
+		cn = tmp2[ 1 ];
+		dn = 1.0;
+		am = u;
+	} else if ( m === 1.0 ) {
+		// A&S table 16.6: limiting case m = 1: hyperbolic functions
+		sn = tanh( u );
+		cn = 1.0 / cosh( u );
+		dn = cn;
+		am = atan( sinh( u ) );
+	} else if ( m < SQRT_EPS ) {
+		// A&S 16.13.1 for small-u approximations. Additionally, compute K at extra cost in
+		// Order to ensure returned values are correct outside the range [0, K].
+		K4 = 4.0 * ellipk( m );
+		u0 = ( ( u % K4 ) + K4 ) % K4;
+		sincos( u0, tmp2, 1, 0 );
+		s = tmp2[ 0 ];
+		c = tmp2[ 1 ];
+		f = 0.25 * m * ( u0 - ( s * c ) );
+		sn = s - ( f * c );
+		cn = c + ( f * s );
+		dn = 1.0 - ( 0.5 * m * s * s );
+		am = u - ( 0.25 * m * ( u - ( s * c ) ) );
+	} else if ( m > 1.0 - SQRT_EPS ) {
+		// A&S 16.15.1 - 16.15.4 for near-unity approximations. Additionally, compute K at
+		// Extra cost so that we may reflect as needed to ensure the returned values
+		// Are correct.
+
+		// Reduce by the half-period 2K, centered about u = 0
+		K2 = ellipk( m ) * 2.0;
+		uK2cen = ( u / K2 ) + 0.5;
+		uK2 = K2 * ( ( uK2cen % 1.0 ) - 0.5 );
+
+		// Flip sn and cn in this range to get the reflections correct. We must be careful
+
+		// About precisely reusing uK2cen in order to get the edge cases corect.
+		uK4 = uK2cen % 4;
+		FLG = uK4 >= 1.0 && uK4 < 2.0;
+
+		sinhu = sinh( uK2 );
+		sechu = 1.0 / cosh( uK2 );
+		tanhu = tanh( uK2 );
+		gdu = ( floor( uK2cen ) * PI ) + atan( sinhu );
+
+		a = 0.25 * ( 1.0 - m );
+		b = a * ( sinhu - ( uK2 * sechu ) );
+		sn = tanhu + ( b * sechu );
+		cn = sechu - ( b * tanhu );
+		dn = sechu + ( a * ( sinhu + ( uK2 * sechu ) ) * tanhu );
+		am = gdu + b;
+
+		if (FLG) {
+			sn = -sn;
+			cn = -cn;
+		}
+	} else {
+		// A&S 16.4.1. Compute using the arithmetic-geometric mean.
+		a = 1.0;
+		b = sqrt( 1.0 - m );
+		N = -1;
+		do {
+			N += 1;
+			if ( N > 8 ) {
+				throw new Error('Overflow in ellipj');
```

Comment:
```suggestion
				throw new Error( 'unexpected error. Encountered overflow during computation.' );
```

Is throwing an error appropriate here? Or should we just return `NaN`s?

Reply:
I would be fine with returning NaN. I'm not aware of any case in which this can happen, though as stdlib algorithms get more sophisticated, it will eventually become necessary to make sure there is a consistent way to convey algorithm failure. If that's something which already exists in stdlib, I'm not personally aware of it.

Reply:
As of right now, for base special math functions, if input is out-of-bounds, etc, we simply return a sentinel value (e.g., `NaN`). I would think we would do the same here (or `infinity` or whatever makes sense). One thing we try to avoid is requiring users to wrap low-level numerics in try/catch blocks.

If you change to `NaN` (or whatever), feel free to add a comment.

Reply:
I logged a debug error and returned NaN in this case. I don't actually know how to test it, as I'm not aware of any circumstance under which this actually happens. I don't know. I think it's avoided by the near-zero and near-unity approximations for m.

Reply:
Seems reasonable.